metadata/metadata.go

```diff
@@ -1476,7 +1476,11 @@ func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, ini
 	if err := resourceNamedSafely(id); err != nil {
 		return nil, err
 	}
-	if existing, err := serv.lookup.Lookup(id); err == nil {
+	existing, err := serv.lookup.Lookup(id)
+	if _, ok := err.(*ResourceNotFound); err != nil && !ok {
```

Comment:
lets change `ok` to `isReasourceError` or something similar. Have to stare at it for a bit rn to figure out the logic with the `ok`